### PR TITLE
Add dependency injection for FXML controllers

### PR DESCRIPTION
**Pull request by Gurgy**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,6 +17,16 @@
               <mainClass>com.github.cypher.Main</mainClass>
           </configuration>
       </plugin>
+
```
Comment (NotTrulyEvil) [inline pom.xml]: What is the purpose of the changes in this file? Packaging/compiling with _mvn package_ still requires you to specify the main class.

Comment (Gurgy) [inline pom.xml]: As specified in the commit message those changes are aimed at compiling the project with JDK 1.8 instead of 1.5 as predefined. The plugin configuration added to this file helps us do that without manually specifying the JDK flags at compile time.

**Comment by Gurgy** [comment]
Not necessary when using afterburner.fx
